pytorch_lightning/__init__.py

```diff
@@ -1,9 +1,22 @@
 """Root package info."""
 
 import logging
+from typing import Any
 
 from pytorch_lightning.__about__ import *  # noqa: F401, F403
 
+DETAIL = 15  # between logging.INFO and logging.DEBUG, used for logging in production use cases
+
+
+def detail(self: Any, message: str, *args: Any, **kwargs: Any) -> None:
+    if self.isEnabledFor(DETAIL):
+        self._log(DETAIL, message, *args, **kwargs)
+
+
+logging.addLevelName(DETAIL, "DETAIL")
```

Comment:
You can test this works as intended by using this file

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/utilities/test_warnings.py

where we assert the stdout state. It is run separate from `pytest`:

https://github.com/PyTorchLightning/pytorch-lightning/blob/1fc046cde28684316323693dfca227dbd270663f/tests/standalone_tests.sh#L61-L62